### PR TITLE
feat: Implement equality = and inequality <> support for BinaryView

### LINKED ISSUE #10996: Implement equality = and inequality <> support for `BinaryView`
### Is your feature request related to a problem or challenge?

Part of https://github.com/apache/datafusion/issues/10918, `[StringViewArray`](https://docs.rs/arrow/latest/arrow/array/type.StringViewArray.html) support in DataFusion

@Weijun-H  added support for `<>` and `!=` for`StringView` in  https://github.com/apache/datafusion/pull/10985. We also need similar support for `BinaryView`

### Describe the solution you'd like

In order to improve performance of these queries we will need the ability to actually compare `StringViewArrays`  to constant values (and likely to each other)

Thus I would like to be able to run 

`BinaryView = scalar`
`BianryView = StringViewColumn`

I basically want to to run the following queries (where table `foo` has `BinaryView` columns)

```sql
> create table foo as values ('Andrew', 'X'), ('Xiangpeng', 'Xiangpeng'), ('Raphael', 'R');
0 row(s) fetched.
Elapsed 0.002 seconds.

> select * from foo where column1 = 'Andrew';
+---------+---------+
| column1 | column2 |
+---------+---------+
| Andrew  | X       |
+---------+---------+
1 row(s) fetched.
Elapsed 0.003 seconds.

> select * from foo where column1 <> 'Andrew';
+-----------+-----------+
| column1   | column2   |
+-----------+-----------+
| Xiangpeng | Xiangpeng |
| Raphael   | R         |
+-----------+-----------+
2 row(s) fetched.
Elapsed 0.001 seconds.

> select * from foo where column1 = column2;
+-----------+-----------+
| column1   | column2   |
+-----------+-----------+
| Xiangpeng | Xiangpeng |
+-----------+-----------+
1 row(s) fetched.
Elapsed 0.002 seconds.

> select * from foo where column1 <> column2;
+---------+---------+
| column1 | column2 |
+---------+---------+
| Andrew  | X       |
| Raphael | R       |
+---------+---------+
2 row(s) fetched.
Elapsed 0.001 seconds.
```

### Describe alternatives you've considered

I suspect we will need to update the coercion logic and maybe also the arrow equality kernels like https://docs.rs/arrow/latest/arrow/compute/kernels/cmp/fn.eq.html 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by PsiACE**
## Which issue does this PR close?

Closes #10996 .

### CODE REVIEW

File: datafusion/expr/src/type_coercion/binary.rs [hunk 1]
```diff
@@ -991,6 +991,9 @@ fn binary_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<DataType>
         (Binary | Utf8, Binary) | (Binary, Utf8) => Some(Binary),
         (LargeBinary | Binary | Utf8 | LargeUtf8, LargeBinary)
         | (LargeBinary, Binary | Utf8 | LargeUtf8) => Some(LargeBinary),
+        (BinaryView, BinaryView) | (BinaryView, Binary) | (Binary, BinaryView) => {
```
Comment (alamb) [inline datafusion/expr/src/type_coercion/binary.rs:994]: 🤔  I wonder if we should also handle `LargeBinary` as well

**Review by alamb — APPROVED** [review]
Thank you @PsiACE  -- very nice

File: datafusion/expr/src/type_coercion/binary.rs (hunk 1 above)
Comment (PsiACE) [reply datafusion/expr/src/type_coercion/binary.rs:994]: We also don't handle `LargeUtf8`. :eyes:

Comment (alamb) [reply datafusion/expr/src/type_coercion/binary.rs:994]: Good call -- I filed https://github.com/apache/datafusion/issues/11023 to track

**Comment by alamb** [comment]
Thanks again @PsiACE

**Pull request merged**